gslib/tracker_file.py

```diff
@@ -554,14 +554,28 @@ def WriteDownloadComponentTrackerFile(tracker_file_name, src_obj_metadata,
 def _WriteTrackerFile(tracker_file_name, data):
   """Creates a tracker file, storing the input data."""
   try:
-    with os.fdopen(os.open(tracker_file_name, os.O_WRONLY | os.O_CREAT, 0o600),
-                   'w') as tf:
+    fd = os.open(tracker_file_name, os.O_WRONLY | os.O_CREAT | os.O_TRUNC,
+                 0o600)
+    with os.fdopen(fd, 'w') as tf:
       tf.write(data)
     return False
   except (IOError, OSError) as e:
     raise RaiseUnwritableTrackerFileException(tracker_file_name, e.strerror)
 
 
+def WriteJsonDataToTrackerFile(tracker_file_name, data):
+  """Create a tracker file and write json data to it.
+
+  Raises:
+    TypeError: If the data is not JSON serializable
+  """
+  try:
+    json_str = json.dumps(data)
+  except TypeError as err:
+    raise RaiseUnwritableTrackerFileException(tracker_file_name, err.strerror)
+  _WriteTrackerFile(tracker_file_name, json_str)
```

Comment:
I believe this should be inside the above `try:` right?

Reply:
I intentionally kept _WriteTrackerFile outside because it has it's own try/except block. I wanted to isolate the errors that could possibly generate from `json.dumps`.